lib/login/ui/app_sign_in.dart

```diff
@@ -8,20 +8,28 @@ import 'package:myecl/login/router.dart';
 import 'package:myecl/login/tools/constants.dart';
 import 'package:myecl/login/ui/auth_page.dart';
 import 'package:myecl/login/ui/components/sign_in_up_bar.dart';
+import 'package:myecl/router.dart';
 import 'package:myecl/tools/constants.dart';
 import 'package:myecl/tools/functions.dart';
-import 'package:myecl/tools/providers/path_forwarding_provider.dart';
+import 'package:myecl/routing/providers/auth_redirect_service_provider.dart';
 import 'package:qlevar_router/qlevar_router.dart';
 
 class AppSignIn extends HookConsumerWidget {
   const AppSignIn({super.key});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final authNotifier = ref.watch(authTokenProvider.notifier);
-    final pathForwarding = ref.read(pathForwardingProvider);
+    final authNotifier = ref.read(authTokenProvider.notifier);
     final controller = ref.watch(backgroundAnimationProvider);
 
+    final forwardedFrom =
+        QR.params[forwardedFromKey]?.toString() ?? AppRouter.root;
+    final redirectPath = ref.watch(
+      authRedirectServiceProvider.select(
+        (service) => service.getRedirect(forwardedFrom),
```

Comment:
The service should take QR.currentPath instead
```suggestion
    final redirectPath = ref.watch(
      authRedirectServiceProvider.select(
        (service) => service.getRedirect(QR.currentPath),
```